config/routes.rb

```diff
@@ -8,4 +18,16 @@
       resources :likes, only: [:new, :create]
     end
   end
+
+  # ROUTES FOR API
+  namespace :api do
+    namespace :v1 do
+      resources :users do
+        resources :posts, format: :json do
+          resources :comments, format: :json
+          resources :likes, format: :json
+        end
+      end
```

Comment:
- [ ] it would be better to add authentication to the API as requested  in the requirements 
- Screenshot of the requirements 
<img width="374" alt="Screenshot 2022-12-15 at 17 27 33" src="https://user-images.githubusercontent.com/61790587/207915445-36ce0dd9-dc9c-4dd4-aceb-ab1105b0929c.png">